src/contexts/SiteLaunchContext.tsx

```diff
@@ -0,0 +1,87 @@
+import { createContext, useContext, useEffect, useState } from "react"
+
+import { useGetSiteLaunchStatus } from "hooks/siteDashboardHooks"
+
+import {
+  SiteLaunchDto,
+  SiteLaunchFrontEndStatus,
+  SiteLaunchStatusProps,
+  SITE_LAUNCH_TASKS_LENGTH,
+} from "types/siteLaunch"
+
+interface SiteLaunchContextProps {
+  setSiteName: (siteName: string) => void
+  siteLaunchStatusProps: SiteLaunchStatusProps
+  setSiteLaunchStatusProps: (
+    siteLaunchStatusProps: SiteLaunchStatusProps
+  ) => void
+}
+
+const SiteLaunchContext = createContext<SiteLaunchContextProps | null>(null)
```

Comment:
referencing your pr description -
I thought you will be using local storage to persist the checklist progress, is that not the case anymore?

Reply:
Uhm regarding our previous discussion, once I fixed the issue I was facing I left it as this for the first cut, but personally no strong opinions on using local storage instead.
Thoughts?

Reply:
agreed, it will either be local storage or ideally will shift state away from FE in the future as a whole. Nat is aware of this, and this limitation will be tackled this in a separate PR!